addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from datetime import datetime, time
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields
+
+from odoo.tools import mute_logger
+
+from odoo.addons.hr_holidays.tests.common import TestHrHolidaysBase
+
+
+class TestAccrualAllocations(TestHrHolidaysBase):
+    def setUp(self):
+        super(TestAccrualAllocations, self).setUp()
+
+        # Make sure we have the rights to create, validate and delete the leaves, leave types and allocations
+        self.LeaveType = self.env['hr.leave.type'].sudo(self.user_hrmanager_id).with_context(tracking_disable=True)
+        self.Allocation = self.env['hr.leave.allocation'].sudo(self.user_hrmanager_id).with_context(tracking_disable=True)
+        self.Leave = self.env['hr.leave'].sudo(self.user_hrmanager_id).with_context(tracking_disable=True)
+
+        self.accrual_type = self.LeaveType.create({
+            'name': 'accrual',
+            'limit': False,
+            'accrual': True,
+        })
+
+        self.unpaid_type = self.LeaveType.create({
+            'name': 'unpaid',
+            'limit': True,
+            'unpaid': True,
+        })
+
+        self.set_employee_create_date(self.employee_emp_id, '2010-02-03 00:00:00')
+        self.set_employee_create_date(self.employee_hruser_id, '2010-02-03 00:00:00')
+
+    def set_employee_create_date(self, id, newdate):
+        """ This method is a hack in order to be able to define/redefine the create_date
+            of the employees.
+            This is done in SQL because ORM does not allow to write onto the create_date field.
+        """
+        cursor = self.env['hr.employee'].browse(id)._cr
+        cursor.execute("""
+                       UPDATE
+                       hr_employee
+                       SET create_date = '%s'
+                       WHERE id = %s
+                       """ % (newdate, id))
+
+    def test_accrual_base_no_leaves(self):
+        alloc_0 = self.Allocation.create({
+            'name': 'Accrual allocation for employee',
+            'employee_id': self.employee_emp_id,
+            'holiday_status_id': self.accrual_type.id,
+            'number_of_days_temp': 0,
+            'number_per_interval': 1,
+            'interval_number': 1,
+            'unit_per_interval': 'days',
+            'interval_unit': 'weeks',
+        })
+
+        alloc_0.action_approve()
+
+        self.Allocation._update_accrual()
+
+        self.assertEqual(alloc_0.number_of_days, 1, 'Employee should have been allocated one leave day')
+
+    def test_accrual_base_leaves(self):
```

Comment:
For all your test, just add a little description of what is the test case in doc string of the test method.